Quick-Start [DA]/MVP/MVP+Makefile.md

```diff
@@ -1 +1,274 @@
+# my_extension--1.0.sql
+```
+
+### Step 3: Create the Extension SQL File
+- Create a file named "my_extension--1.0.sql".
+  - This file contains the SQL statements required to install or uninstall your extension.
+
+```bash
+$ touch my_extension--1.0.sql
+```
+
+- Open the "my_extension--1.0.sql" file and add the following content:
+
+```sql
+-- extension--1.0.sql
+-- my_extension--1.0.sql
+-- Create necessary objects for version 1.0
+CREATE TABLE my_table (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(100) NOT NULL
+);
+```
+
+### Step 4: Create the Makefile
+- Create a Makefile in the "extension" directory. 
+  - The Makefile is used to define the build process. It specifies the module name, object files, extension name, and SQL files. The PG_CONFIG variable retrieves the path to the pg_config utility, which is used to get the necessary configuration for building the extension.
+
+```bash
+$ touch Makefile
+```
+- Open the Makefile and add the following content:
+<!--Problem: https://dba.stackexchange.com/questions/328624/makefile-issue-in-creating-postgres-extension
+Had an issue with the make install command:
+```makefile
+DATA = my_extension--1.0.sql
+
+PG_CONFIG  ?= pg_config
+PGXS := $(shell $(PG_CONFIG) --pgxs)
+include $(PGXS)
+```
+Solution: Makefile is lacking the entry `EXTENSION = my_extension`
+Reason: That's why `make install` doesn't copy the `*.control` file, and that's why `CREATE EXTENSION` fails.
+
+NOTE: make: Nothing to be done for 'all' 
+Makefile doesn't have any MODULE or other binary to build. You can go directly to the make install step
+-->
+
+```makefile
+EXTENSION = my_extension
+DATA = my_extension--1.0.sql
+
+PG_CONFIG  ?= pg_config
+PGXS := $(shell $(PG_CONFIG) --pgxs)
+include $(PGXS)
+```
+
+### Step 5: Build and Install the Extension
+- To build and install the extension, run the following commands:
+
+```bash
+$ make
+$ make install
+```
+- The "make" command compiles the extension code using the Makefile
+- The "make install" command copies the necessary files to the appropriate PostgreSQL directories.
+
+### Step 6: Create the Extension in PostgreSQL
+- Connect to your PostgreSQL database and execute the following SQL statement to create the extension:
+
+```sql
+CREATE EXTENSION IF NOT EXITS "my_extension";
+```
+- The CREATE EXTENSION statement is executed in PostgreSQL to create the extension.
+
+## Implementation of the Extension:
+Once you have successfully implemented the `my_extension` in your PostgreSQL database, you can use it as follows:
+
+### 1. Query the `my_table` table:
+   - You can query the `my_table` table like any other table in your database. For example, you can retrieve all rows from the table using the following SQL query:
+   ```sql
+   SELECT * FROM my_table;
+   ```
+#### Output:
+```sql
+id | name
+----+------
+(0 rows)
+```
+Remember to replace `my_table` with the actual table name you have chosen, and adjust the function call as needed based on your specific implementation.
```

Comment:
I don't understand this sentence, since you didn't have the reader change the table name, nor is there a function call. Leftover from a previous doc?

Reply:
the first part of the point is faulty i will look into it

yup, the function call is something I have to fix.
When I fix the function, the 2nd part of the line will make sense.